src/status_im2/contexts/wallet/send/select_asset/view.cljs

```diff
@@ -0,0 +1,136 @@
+(ns status-im2.contexts.wallet.send.select-asset.view
+  (:require
+    [clojure.string :as string]
+    [quo.core :as quo]
+    [quo.foundations.resources :as quo.resources]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]
+    [react-native.safe-area :as safe-area]
+    [reagent.core :as reagent]
+    [status-im2.contexts.wallet.send.select-asset.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def tabs-data
+  [{:id :tab/assets :label (i18n/label :t/assets) :accessibility-label :assets-tab}
+   {:id :tab/collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}])
+
+(defn- network-names
+  [balances-per-chain]
+  (mapv (fn [chain-id-keyword]
+          (let [chain-id-str (name chain-id-keyword)
+                chain-id     (js/parseInt chain-id-str)]
+            (case chain-id
+              10    {:source (quo.resources/get-network :optimism)}
+              42161 {:source (quo.resources/get-network :arbitrum)}
+              5     {:source (quo.resources/get-network :ethereum)}
+              1     {:source (quo.resources/get-network :ethereum)}
+              :unknown))) ; Default case if the chain-id is not recognized
+        (keys balances-per-chain)))
+
+(defn- asset-component
+  []
+  (fn [token _ _ _]
+    (let [on-press                #(js/alert "Not implemented yet")
+          total-balance           (reduce +
+                                          (map #(js/parseFloat (:balance %))
+                                               (vals (:balancesPerChain token))))
+          total-balance-formatted (.toFixed total-balance 2)
+          currency                :usd
+          currency-symbol         "$"
+          price-fiat-currency     (get-in token [:marketValuesPerCurrency currency :price])
+          balance-fiat            (* total-balance price-fiat-currency)
+          balance-fiat-formatted  (.toFixed balance-fiat 2)
+          networks-list           (network-names (:balancesPerChain token))]
```

Comment:
`networks-list (rf/sub [:wallet/network-details]`

Reply:
Actually this filters the chains per token which may differ for each token (i.e. one token available in Mainnet and Optimism, other in Mainnet, Optimism and Arbitrum, etc). Would the that subscription work for that purpose?

Reply:
nope, so what we probably have to do is make a new sub for this. Use the sub for network details to get the relevant networks and then get the respective tokens for that. Pretty much what you have done here but we abstract to a sub I guess :)

Reply:
I simplified the approach in a single sub 👍